### PR TITLE
fix(router): make setUpLocationChangeListener idempotent

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

**Comment by robwormald** [comment]
LGTM

File: modules/@angular/router/upgrade.ts [hunk 1]
```diff
@@ -40,10 +40,20 @@ export const RouterUpgradeInitializer = {
...
+/**
+ * @internal
+ */
 export function initialRouterNavigation(
     ngUpgrade: UpgradeModule, ref: ApplicationRef, preloader: RouterPreloader,
     opts: ExtraOptions): Function {
   return () => {
+    if (!ngUpgrade.$injector) {
+      throw new Error(`
+        RouterUpgradeInitializer can be used only after UpgradeModule.bootrstrap has been called.
```
Comment (vicb) [inline modules/@angular/router/upgrade.ts]: "bootrstrap" (typo) * 2

File: modules/@angular/router/upgrade.ts [hunk 2]
```diff
@@ -52,17 +62,27 @@ export function initialRouterNavigation(
...
-          const url = document.createElement('a');
-          url.href = next;
-          router.navigateByUrl(url.pathname);
-        });
+    setUpLocationSync(ngUpgrade);
   };
+}
+
+/**
+ * @whatItDoes Set ups a location synchronization.
```
Comment (vicb) [inline modules/@angular/router/upgrade.ts]: "sets up"

**Comment by alxhub** [comment]
Rebased and merged as c767df0.

**Pull request closed without merging**